test/e2e/gateway/gateway_reporting.go

```diff
@@ -0,0 +1,81 @@
+package redundancy
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	subv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	subFramework "github.com/submariner-io/submariner/test/e2e/framework"
+)
+
+var _ = Describe("[gateway] Gateway reporting tests", func() {
+	f := subFramework.NewFramework("gateway-reporting")
+
+	When("any gateway node is configured", func() {
+		It("should be reported to the Gateway API", func() {
+			testBasicGatewayReporting(f)
+		})
+	})
+
+	When("multiple gateway nodes are configured", func() {
+		It("only one is active", func() {
+			testOnlyOneActiveGateway(f)
+		})
+	})
+
+})
+
+func testBasicGatewayReporting(f *subFramework.Framework) {
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+
+	By(fmt.Sprintf("Sanity check - ensuring there's exactly one gateway node on %q", clusterAName))
+	gatewayNodes := f.FindNodesByGatewayLabel(framework.ClusterA, true)
+	Expect(gatewayNodes).To(HaveLen(1), fmt.Sprintf("Expected only one gateway node on %q", clusterAName))
+
+	By(fmt.Sprintf("Ensuring that the gateway reports as active %q", clusterAName))
+	gatewayClient := subFramework.SubmarinerClients[framework.ClusterA].SubmarinerV1().Gateways(
+		framework.TestContext.SubmarinerNamespace)
+	gwList, err := gatewayClient.List(v1meta.ListOptions{LabelSelector: subv1.HAStatusActiveSelector})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(gwList.Items).To(HaveLen(1))
```

Comment:
Can't this call `f.AwaitForGatewayWithStatus`?

Reply:
I also wanted to test that the label selection worked properly, I can separate in an additional test with that specific intent.

Reply:
I had a previous comment in the production code asking why you were duplicating the HA status in a label.  I see now why you did it.  But we don't really need the added complexity of a duplicated label just for test code - we can just filter by the `HAStatus` field.

Reply:
No, also for administrators, they can use the label to get (from cmdline, or API) the active gateway, I think it's a valide and helpful use case.

Reply:
I don't see why. One can filter on fields as well.  But if labels are easier then just store as a label  but not both although I think it makes more sense as a field.

Reply:
Ok, my concern on usability is very well handled by the proposal from @Oats87 to make the ha-status printable:

```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: gateways.submariner.io
spec:
  conversion:
    strategy: None
  group: submariner.io
  names:
    kind: Gateway
    listKind: GatewayList
    plural: gateways
    singular: gateway
  scope: Namespaced
  version: v1
  versions:
  - name: v1
    served: true
    storage: true
  additionalPrinterColumns:
  - name: ha-status 
    type: string
    description: High Availability Status of the Gateway 
    JSONPath: .status.haStatus

```

Now:

```
$ kubectl get Gateways -A --context cluster2
NAMESPACE             NAME               HA-STATUS
submariner-operator   cluster2-worker2   active
```
I removed the labels :-)